// ionic-angular/components/list/list.ngfactory.js
/**
 * This file is generated by the Angular 2 template compiler.
 * Do not edit.
 */
/* tslint:disable */
import * as import0 from 'ionic-angular/components/list/list';
import * as import1 from '@angular/core/src/change_detection/change_detection';
import * as import2 from '@angular/core/src/linker/view_utils';
export var Wrapper_List = (function () {
    function Wrapper_List(p0, p1, p2, p3) {
        this.changed = false;
        this._mode = import1.UNINITIALIZED;
        this._sliding = import1.UNINITIALIZED;
        this.context = new import0.List(p0, p1, p2, p3);
    }
    Wrapper_List.prototype.check_mode = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import2.checkBinding(throwOnChange, this._mode, currValue))) {
            this.changed = true;
            this.context.mode = currValue;
            this._mode = currValue;
        }
    };
    Wrapper_List.prototype.check_sliding = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import2.checkBinding(throwOnChange, this._sliding, currValue))) {
            this.changed = true;
            this.context.sliding = currValue;
            this._sliding = currValue;
        }
    };
    Wrapper_List.prototype.detectChangesInternal = function (view, el, throwOnChange) {
        var changed = this.changed;
        this.changed = false;
        return changed;
    };
    return Wrapper_List;
}());
