// ionic-angular/components/item/item.ngfactory.js
/**
 * This file is generated by the Angular 2 template compiler.
 * Do not edit.
 */
/* tslint:disable */
var __extends = (this && this.__extends) || function (d, b) {
    for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p];
    function __() { this.constructor = d; }
    d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
};
import * as import0 from 'ionic-angular/components/item/item';
import * as import1 from '@angular/core/src/change_detection/change_detection';
import * as import2 from '@angular/core/src/linker/view_utils';
import * as import3 from '@angular/core/src/linker/view';
import * as import5 from '@angular/core/src/linker/element';
import * as import6 from '@angular/core/src/linker/query_list';
import * as import8 from '@angular/core/src/linker/view_type';
import * as import9 from 'ionic-angular/util/form';
import * as import10 from 'ionic-angular/config/config';
import * as import11 from '@angular/core/src/linker/element_ref';
import * as import12 from '@angular/core/src/metadata/view';
import * as import13 from '@angular/core/src/linker/component_factory';
import * as import14 from '../../../@angular/common/src/directives/ng_if.ngfactory';
import * as import15 from './item-reorder.ngfactory';
import * as import16 from '@angular/core/src/linker/template_ref';
import * as import17 from '@angular/common/src/directives/ng_if';
import * as import18 from 'ionic-angular/components/item/item-reorder';
import * as import19 from '../label/label.ngfactory';
import * as import20 from 'ionic-angular/components/label/label';
export var Wrapper_Item = (function () {
    function Wrapper_Item(p0, p1, p2, p3) {
        this.changed = false;
        this._color = import1.UNINITIALIZED;
        this._mode = import1.UNINITIALIZED;
        this.context = new import0.Item(p0, p1, p2, p3);
    }
    Wrapper_Item.prototype.check_color = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import2.checkBinding(throwOnChange, this._color, currValue))) {
            this.changed = true;
            this.context.color = currValue;
            this._color = currValue;
        }
    };
    Wrapper_Item.prototype.check_mode = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import2.checkBinding(throwOnChange, this._mode, currValue))) {
            this.changed = true;
            this.context.mode = currValue;
            this._mode = currValue;
        }
    };
    Wrapper_Item.prototype.detectChangesInternal = function (view, el, throwOnChange) {
        var changed = this.changed;
        this.changed = false;
        return changed;
    };
    return Wrapper_Item;
}());
export var Wrapper_ItemContent = (function () {
    function Wrapper_ItemContent() {
        this.changed = false;
        this.context = new import0.ItemContent();
    }
    Wrapper_ItemContent.prototype.detectChangesInternal = function (view, el, throwOnChange) {
        var changed = this.changed;
        this.changed = false;
        return changed;
    };
    return Wrapper_ItemContent;
}());
export var Wrapper_ItemGroup = (function () {
    function Wrapper_ItemGroup() {
        this.changed = false;
        this.context = new import0.ItemGroup();
    }
    Wrapper_ItemGroup.prototype.detectChangesInternal = function (view, el, throwOnChange) {
        var changed = this.changed;
        this.changed = false;
        return changed;
    };
    return Wrapper_ItemGroup;
}());
var renderType_Item_Host = null;
var _View_Item_Host0 = (function (_super) {
    __extends(_View_Item_Host0, _super);
    function _View_Item_Host0(viewUtils, parentInjector, declarationEl) {
        _super.call(this, _View_Item_Host0, renderType_Item_Host, import8.ViewType.HOST, viewUtils, parentInjector, declarationEl, import1.ChangeDetectorStatus.CheckAlways);
    }
    _View_Item_Host0.prototype.createInternal = function (rootSelector) {
        this._el_0 = this.selectOrCreateHostElement('ion-list-header', rootSelector, null);
        this.renderer.setElementAttribute(this._el_0, 'class', 'item');
        this._appEl_0 = new import5.AppElement(0, null, this, this._el_0);
        var compView_0 = viewFactory_Item0(this.viewUtils, this.injector(0), this._appEl_0);
        this._Item_0_4 = new Wrapper_Item(this.parentInjector.get(import9.Form), this.parentInjector.get(import10.Config), new import11.ElementRef(this._el_0), this.renderer);
        this._query_Label_0_0 = new import6.QueryList();
        this._query_Button_0_1 = new import6.QueryList();
        this._query_Icon_0_2 = new import6.QueryList();
        this._appEl_0.initComponent(this._Item_0_4.context, [], compView_0);
        this._query_Label_0_0.reset([]);
        this._Item_0_4.context.contentLabel = this._query_Label_0_0.first;
        compView_0.create(this._Item_0_4.context, this.projectableNodes, null);
        this.init([].concat([this._el_0]), [this._el_0], [], []);
        return this._appEl_0;
    };
    _View_Item_Host0.prototype.injectorGetInternal = function (token, requestNodeIndex, notFoundResult) {
        if (((token === import0.Item) && (0 === requestNodeIndex))) {
            return this._Item_0_4.context;
        }
        return notFoundResult;
    };
    _View_Item_Host0.prototype.detectChangesInternal = function (throwOnChange) {
        if (this._Item_0_4.detectChangesInternal(this, this._el_0, throwOnChange)) {
            this._appEl_0.componentView.markAsCheckOnce();
        }
        this.detectContentChildrenChanges(throwOnChange);
        if (!throwOnChange) {
            if (this._query_Button_0_1.dirty) {
                this._query_Button_0_1.reset([]);
                this._Item_0_4.context._buttons = this._query_Button_0_1;
                this._query_Button_0_1.notifyOnChanges();
            }
            if (this._query_Icon_0_2.dirty) {
                this._query_Icon_0_2.reset([]);
                this._Item_0_4.context._icons = this._query_Icon_0_2;
                this._query_Icon_0_2.notifyOnChanges();
            }
            if ((this.numberOfChecks === 0)) {
                this._Item_0_4.context.ngAfterContentInit();
            }
        }
        this.detectViewChildrenChanges(throwOnChange);
    };
    return _View_Item_Host0;
}(import3.AppView));
function viewFactory_Item_Host0(viewUtils, parentInjector, declarationEl) {
    if ((renderType_Item_Host === null)) {
        (renderType_Item_Host = viewUtils.createRenderComponentType('', 0, import12.ViewEncapsulation.None, [], {}));
    }
    return new _View_Item_Host0(viewUtils, parentInjector, declarationEl);
}
export var ItemNgFactory = new import13.ComponentFactory('ion-list-header,ion-item,[ion-item],ion-item-divider', viewFactory_Item_Host0, import0.Item);
var styles_Item = [];
var renderType_Item = null;
var _View_Item0 = (function (_super) {
    __extends(_View_Item0, _super);
    function _View_Item0(viewUtils, parentInjector, declarationEl) {
        _super.call(this, _View_Item0, renderType_Item, import8.ViewType.COMPONENT, viewUtils, parentInjector, declarationEl, import1.ChangeDetectorStatus.CheckOnce);
    }
    _View_Item0.prototype.createInternal = function (rootSelector) {
        var parentRenderNode = this.renderer.createViewRoot(this.declarationAppElement.nativeElement);
        this._viewQuery_Label_0 = new import6.QueryList();
        this.renderer.projectNodes(parentRenderNode, import2.flattenNestedViewRenderNodes(this.projectableNodes[0]));
        this._el_0 = this.renderer.createElement(parentRenderNode, 'div', null);
        this.renderer.setElementAttribute(this._el_0, 'class', 'item-inner');
        this._el_1 = this.renderer.createElement(this._el_0, 'div', null);
        this.renderer.setElementAttribute(this._el_1, 'class', 'input-wrapper');
        this.renderer.projectNodes(this._el_1, import2.flattenNestedViewRenderNodes(this.projectableNodes[1]));
        this._anchor_2 = this.renderer.createTemplateAnchor(this._el_1, null);
        this._appEl_2 = new import5.AppElement(2, 1, this, this._anchor_2);
        this._TemplateRef_2_5 = new import16.TemplateRef_(this._appEl_2, viewFactory_Item1);
        this._NgIf_2_6 = new import14.Wrapper_NgIf(this._appEl_2.vcRef, this._TemplateRef_2_5);
        this.renderer.projectNodes(this._el_1, import2.flattenNestedViewRenderNodes(this.projectableNodes[3]));
        this.renderer.projectNodes(this._el_0, import2.flattenNestedViewRenderNodes(this.projectableNodes[4]));
        this._el_3 = this.renderer.createElement(this._el_0, 'ion-reorder', null);
        this._appEl_3 = new import5.AppElement(3, 0, this, this._el_3);
        var compView_3 = import15.viewFactory_Reorder0(this.viewUtils, this.injector(3), this._appEl_3);
        this._Reorder_3_4 = new import15.Wrapper_Reorder(this.parentInjector.get(import0.Item), new import11.ElementRef(this._el_3));
        this._appEl_3.initComponent(this._Reorder_3_4.context, [], compView_3);
        compView_3.create(this._Reorder_3_4.context, [], null);
        this._el_4 = this.renderer.createElement(parentRenderNode, 'div', null);
        this.renderer.setElementAttribute(this._el_4, 'class', 'button-effect');
        var disposable_0 = this.renderer.listen(this._el_3, 'click', this.eventHandler(this._handle_click_3_0.bind(this)));
        this.init([], [
            this._el_0,
            this._el_1,
            this._anchor_2,
            this._el_3,
            this._el_4
        ], [disposable_0], []);
        return null;
    };
    _View_Item0.prototype.injectorGetInternal = function (token, requestNodeIndex, notFoundResult) {
        if (((token === import16.TemplateRef) && (2 === requestNodeIndex))) {
            return this._TemplateRef_2_5;
        }
        if (((token === import17.NgIf) && (2 === requestNodeIndex))) {
            return this._NgIf_2_6.context;
        }
        if (((token === import18.Reorder) && (3 === requestNodeIndex))) {
            return this._Reorder_3_4.context;
        }
        return notFoundResult;
    };
    _View_Item0.prototype.detectChangesInternal = function (throwOnChange) {
        var currVal_0 = this.context._viewLabel;
        this._NgIf_2_6.check_ngIf(currVal_0, throwOnChange, false);
        this._NgIf_2_6.detectChangesInternal(this, this._anchor_2, throwOnChange);
        this._Reorder_3_4.detectChangesInternal(this, this._el_3, throwOnChange);
        this.detectContentChildrenChanges(throwOnChange);
        this.detectViewChildrenChanges(throwOnChange);
        if (!throwOnChange) {
            if (this._viewQuery_Label_0.dirty) {
                this._viewQuery_Label_0.reset([this._appEl_2.mapNestedViews(_View_Item1, function (nestedView) {
                        return [nestedView._Label_0_3.context];
                    })]);
                this.context.viewLabel = this._viewQuery_Label_0.first;
            }
        }
    };
    _View_Item0.prototype._handle_click_3_0 = function ($event) {
        this._appEl_3.componentView.markPathToRootAsCheckOnce();
        var pd_0 = (this._Reorder_3_4.context.onClick($event) !== false);
        return (true && pd_0);
    };
    return _View_Item0;
}(import3.AppView));
export function viewFactory_Item0(viewUtils, parentInjector, declarationEl) {
    if ((renderType_Item === null)) {
        (renderType_Item = viewUtils.createRenderComponentType('', 5, import12.ViewEncapsulation.None, styles_Item, {}));
    }
    return new _View_Item0(viewUtils, parentInjector, declarationEl);
}
var _View_Item1 = (function (_super) {
    __extends(_View_Item1, _super);
    function _View_Item1(viewUtils, parentInjector, declarationEl) {
        _super.call(this, _View_Item1, renderType_Item, import8.ViewType.EMBEDDED, viewUtils, parentInjector, declarationEl, import1.ChangeDetectorStatus.CheckAlways);
    }
    _View_Item1.prototype.createInternal = function (rootSelector) {
        this._el_0 = this.renderer.createElement(null, 'ion-label', null);
        this._Label_0_3 = new import19.Wrapper_Label(this.parent.parentInjector.get(import10.Config), new import11.ElementRef(this._el_0), this.renderer, null, null, null, null);
        this.renderer.projectNodes(this._el_0, import2.flattenNestedViewRenderNodes(this.projectableNodes[2]));
        this.init([].concat([this._el_0]), [this._el_0], [], []);
        return null;
    };
    _View_Item1.prototype.injectorGetInternal = function (token, requestNodeIndex, notFoundResult) {
        if (((token === import20.Label) && (0 === requestNodeIndex))) {
            return this._Label_0_3.context;
        }
        return notFoundResult;
    };
    _View_Item1.prototype.detectChangesInternal = function (throwOnChange) {
        this._Label_0_3.detectChangesInternal(this, this._el_0, throwOnChange);
        this.detectContentChildrenChanges(throwOnChange);
        this.detectViewChildrenChanges(throwOnChange);
    };
    _View_Item1.prototype.dirtyParentQueriesInternal = function () {
        this.parent._viewQuery_Label_0.setDirty();
    };
    return _View_Item1;
}(import3.AppView));
function viewFactory_Item1(viewUtils, parentInjector, declarationEl) {
    return new _View_Item1(viewUtils, parentInjector, declarationEl);
}
