// ionic-angular/components/menu/menu.ngfactory.js
/**
 * This file is generated by the Angular 2 template compiler.
 * Do not edit.
 */
/* tslint:disable */
var __extends = (this && this.__extends) || function (d, b) {
    for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p];
    function __() { this.constructor = d; }
    d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
};
import * as import0 from 'ionic-angular/components/menu/menu';
import * as import1 from '@angular/core/src/change_detection/change_detection';
import * as import2 from '@angular/core/src/linker/view_utils';
import * as import3 from '@angular/core/src/linker/view';
import * as import5 from '@angular/core/src/linker/element';
import * as import6 from '@angular/core/src/linker/query_list';
import * as import8 from '@angular/core/src/linker/view_type';
import * as import9 from 'ionic-angular/components/menu/menu-controller';
import * as import10 from '@angular/core/src/linker/element_ref';
import * as import11 from 'ionic-angular/config/config';
import * as import12 from 'ionic-angular/platform/platform';
import * as import13 from 'ionic-angular/util/keyboard';
import * as import14 from '@angular/core/src/zone/ng_zone';
import * as import15 from 'ionic-angular/gestures/gesture-controller';
import * as import16 from '@angular/core/src/metadata/view';
import * as import17 from '@angular/core/src/linker/component_factory';
import * as import18 from '../backdrop/backdrop.ngfactory';
import * as import19 from 'ionic-angular/components/backdrop/backdrop';
export var Wrapper_Menu = (function () {
    function Wrapper_Menu(p0, p1, p2, p3, p4, p5, p6, p7) {
        this.changed = false;
        this._content = import1.UNINITIALIZED;
        this._id = import1.UNINITIALIZED;
        this._side = import1.UNINITIALIZED;
        this._type = import1.UNINITIALIZED;
        this._enabled = import1.UNINITIALIZED;
        this._swipeEnabled = import1.UNINITIALIZED;
        this._persistent = import1.UNINITIALIZED;
        this._maxEdgeStart = import1.UNINITIALIZED;
        this.context = new import0.Menu(p0, p1, p2, p3, p4, p5, p6, p7);
    }
    Wrapper_Menu.prototype.check_content = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import2.checkBinding(throwOnChange, this._content, currValue))) {
            this.changed = true;
            this.context.content = currValue;
            this._content = currValue;
        }
    };
    Wrapper_Menu.prototype.check_id = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import2.checkBinding(throwOnChange, this._id, currValue))) {
            this.changed = true;
            this.context.id = currValue;
            this._id = currValue;
        }
    };
    Wrapper_Menu.prototype.check_side = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import2.checkBinding(throwOnChange, this._side, currValue))) {
            this.changed = true;
            this.context.side = currValue;
            this._side = currValue;
        }
    };
    Wrapper_Menu.prototype.check_type = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import2.checkBinding(throwOnChange, this._type, currValue))) {
            this.changed = true;
            this.context.type = currValue;
            this._type = currValue;
        }
    };
    Wrapper_Menu.prototype.check_enabled = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import2.checkBinding(throwOnChange, this._enabled, currValue))) {
            this.changed = true;
            this.context.enabled = currValue;
            this._enabled = currValue;
        }
    };
    Wrapper_Menu.prototype.check_swipeEnabled = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import2.checkBinding(throwOnChange, this._swipeEnabled, currValue))) {
            this.changed = true;
            this.context.swipeEnabled = currValue;
            this._swipeEnabled = currValue;
        }
    };
    Wrapper_Menu.prototype.check_persistent = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import2.checkBinding(throwOnChange, this._persistent, currValue))) {
            this.changed = true;
            this.context.persistent = currValue;
            this._persistent = currValue;
        }
    };
    Wrapper_Menu.prototype.check_maxEdgeStart = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import2.checkBinding(throwOnChange, this._maxEdgeStart, currValue))) {
            this.changed = true;
            this.context.maxEdgeStart = currValue;
            this._maxEdgeStart = currValue;
        }
    };
    Wrapper_Menu.prototype.detectChangesInternal = function (view, el, throwOnChange) {
        var changed = this.changed;
        this.changed = false;
        if (!throwOnChange) {
            if ((view.numberOfChecks === 0)) {
                this.context.ngOnInit();
            }
        }
        return changed;
    };
    return Wrapper_Menu;
}());
var renderType_Menu_Host = null;
var _View_Menu_Host0 = (function (_super) {
    __extends(_View_Menu_Host0, _super);
    function _View_Menu_Host0(viewUtils, parentInjector, declarationEl) {
        _super.call(this, _View_Menu_Host0, renderType_Menu_Host, import8.ViewType.HOST, viewUtils, parentInjector, declarationEl, import1.ChangeDetectorStatus.CheckAlways);
    }
    _View_Menu_Host0.prototype.createInternal = function (rootSelector) {
        this._el_0 = this.selectOrCreateHostElement('ion-menu', rootSelector, null);
        this.renderer.setElementAttribute(this._el_0, 'role', 'navigation');
        this._appEl_0 = new import5.AppElement(0, null, this, this._el_0);
        var compView_0 = viewFactory_Menu0(this.viewUtils, this.injector(0), this._appEl_0);
        this._Menu_0_4 = new Wrapper_Menu(this.parentInjector.get(import9.MenuController), new import10.ElementRef(this._el_0), this.parentInjector.get(import11.Config), this.parentInjector.get(import12.Platform), this.renderer, this.parentInjector.get(import13.Keyboard), this.parentInjector.get(import14.NgZone), this.parentInjector.get(import15.GestureController));
        this._query_Content_0_0 = new import6.QueryList();
        this._appEl_0.initComponent(this._Menu_0_4.context, [], compView_0);
        this._query_Content_0_0.reset([]);
        this._Menu_0_4.context.menuContent = this._query_Content_0_0.first;
        compView_0.create(this._Menu_0_4.context, this.projectableNodes, null);
        this.init([].concat([this._el_0]), [this._el_0], [], []);
        return this._appEl_0;
    };
    _View_Menu_Host0.prototype.injectorGetInternal = function (token, requestNodeIndex, notFoundResult) {
        if (((token === import0.Menu) && (0 === requestNodeIndex))) {
            return this._Menu_0_4.context;
        }
        return notFoundResult;
    };
    _View_Menu_Host0.prototype.detectChangesInternal = function (throwOnChange) {
        if (this._Menu_0_4.detectChangesInternal(this, this._el_0, throwOnChange)) {
            this._appEl_0.componentView.markAsCheckOnce();
        }
        this.detectContentChildrenChanges(throwOnChange);
        this.detectViewChildrenChanges(throwOnChange);
    };
    _View_Menu_Host0.prototype.destroyInternal = function () {
        this._Menu_0_4.context.ngOnDestroy();
    };
    return _View_Menu_Host0;
}(import3.AppView));
function viewFactory_Menu_Host0(viewUtils, parentInjector, declarationEl) {
    if ((renderType_Menu_Host === null)) {
        (renderType_Menu_Host = viewUtils.createRenderComponentType('', 0, import16.ViewEncapsulation.None, [], {}));
    }
    return new _View_Menu_Host0(viewUtils, parentInjector, declarationEl);
}
export var MenuNgFactory = new import17.ComponentFactory('ion-menu', viewFactory_Menu_Host0, import0.Menu);
var styles_Menu = [];
var renderType_Menu = null;
var _View_Menu0 = (function (_super) {
    __extends(_View_Menu0, _super);
    function _View_Menu0(viewUtils, parentInjector, declarationEl) {
        _super.call(this, _View_Menu0, renderType_Menu, import8.ViewType.COMPONENT, viewUtils, parentInjector, declarationEl, import1.ChangeDetectorStatus.CheckOnce);
    }
    _View_Menu0.prototype.createInternal = function (rootSelector) {
        var parentRenderNode = this.renderer.createViewRoot(this.declarationAppElement.nativeElement);
        this._viewQuery_Backdrop_0 = new import6.QueryList();
        this._el_0 = this.renderer.createElement(parentRenderNode, 'div', null);
        this.renderer.setElementAttribute(this._el_0, 'class', 'menu-inner');
        this.renderer.projectNodes(this._el_0, import2.flattenNestedViewRenderNodes(this.projectableNodes[0]));
        this._el_1 = this.renderer.createElement(parentRenderNode, 'ion-backdrop', null);
        this.renderer.setElementAttribute(this._el_1, 'disable-activated', '');
        this.renderer.setElementAttribute(this._el_1, 'disableScroll', 'false');
        this.renderer.setElementAttribute(this._el_1, 'role', 'presentation');
        this.renderer.setElementAttribute(this._el_1, 'tappable', '');
        this._Backdrop_1_3 = new import18.Wrapper_Backdrop(this.parentInjector.get(import15.GestureController), new import10.ElementRef(this._el_1), this.renderer);
        this._viewQuery_Backdrop_0.reset([this._Backdrop_1_3.context]);
        this.context.backdrop = this._viewQuery_Backdrop_0.first;
        this.init([], [
            this._el_0,
            this._el_1
        ], [], []);
        return null;
    };
    _View_Menu0.prototype.injectorGetInternal = function (token, requestNodeIndex, notFoundResult) {
        if (((token === import19.Backdrop) && (1 === requestNodeIndex))) {
            return this._Backdrop_1_3.context;
        }
        return notFoundResult;
    };
    _View_Menu0.prototype.detectChangesInternal = function (throwOnChange) {
        var currVal_0 = 'false';
        this._Backdrop_1_3.check_disableScroll(currVal_0, throwOnChange, false);
        this._Backdrop_1_3.detectChangesInternal(this, this._el_1, throwOnChange);
        this.detectContentChildrenChanges(throwOnChange);
        this.detectViewChildrenChanges(throwOnChange);
    };
    _View_Menu0.prototype.destroyInternal = function () {
        this._Backdrop_1_3.context.ngOnDestroy();
    };
    return _View_Menu0;
}(import3.AppView));
export function viewFactory_Menu0(viewUtils, parentInjector, declarationEl) {
    if ((renderType_Menu === null)) {
        (renderType_Menu = viewUtils.createRenderComponentType('', 1, import16.ViewEncapsulation.None, styles_Menu, {}));
    }
    return new _View_Menu0(viewUtils, parentInjector, declarationEl);
}
